source/adios2/core/IO.cpp

```diff
@@ -45,6 +45,8 @@ void IO::SetIOMode(const IOMode ioMode) { m_IOMode = ioMode; };
 
 void IO::SetParameters(const Params &parameters) { m_Parameters = parameters; }
 
+const Params &IO::GetParameters() { return m_Parameters; }
```

Comment:
functions don't decide constness of their return types.  intel compiler will issue those.  And no, you can't make the function const when returning an internal reference... just try if you're not convinced. 

const Foo foo = GetFoo();
Foo foo = GetFoo();